data/models/vegetarianexpe.py

```diff
@@ -0,0 +1,438 @@
+from django.db import models
+from django.core.validators import MinValueValidator, MaxValueValidator
+from data.fields import ChoiceArrayField
+from .canteen import Canteen
+
+
+class VegetarianExpe(models.Model):
+    class Meta:
+        verbose_name = "expérimentation repas végétariens quotidiens"
+        verbose_name_plural = "expérimentations repas végétariens quotidiens"
+
+    class WasteEvolution(models.TextChoices):
+        HIGHER = "higher", "Oui, il y a plus de gaspillage"
+        LOWER = "lower", "Non, il y a moins de gaspillage"
+        SAME = "same", "Pas de différence notable"
+
+    class CostEvolution(models.TextChoices):
+        HIGHER = "higher", "Le coût a augmenté"
+        LOWER = "lower", "Le coût a diminué"
+        SAME = "same", "Pas de différence notable"
+
+    class AttendanceEvolution(models.TextChoices):
+        HIGHER = "higher", "La fréquentation a augmenté"
+        LOWER = "lower", "La fréquentation a diminué"
+        SAME = "same", "Pas de différence notable"
+
+    class SatisfactionReasons(models.TextChoices):
+        CHOICE = "choice", "Liberté de choix (régime, culte...)"
+        TASTE = "taste", "Goût et texture"
+        NOVELTY = "novelty", "Nouveauté"
+        VARIETY = "variety", "Variété des recettes"
+        IGNORANCE = "ignorance", "Méconnaissance"
+        REJECT = "reject", "Opposition de principe"
+        HEALTH = "health", "Impact sur la santé"
+        ENVIRONMENT = "environment", "Impact sur l'environnement"
+
+    creation_date = models.DateTimeField(auto_now_add=True)
+    modification_date = models.DateTimeField(auto_now=True)
+
+    canteen = models.ForeignKey(Canteen, on_delete=models.CASCADE, verbose_name="cantine")
+
+    has_daily_vegetarian_offer = models.BooleanField(
+        default=False, verbose_name="a déjà mis en place l'option végétarienne quotidienne"
+    )
+
+    # Start Date
+    daily_vegetarian_offer_start_date = models.DateField(
+        null=True, blank=True, verbose_name="date de début de l'offre végétarienne quotidienne"
+    )
+    experimentation_start_date = models.DateField(
+        null=True, blank=True, verbose_name="date de début d'expérimentation"
+    )
+
+    # ------------- T0
+
+    vegetarian_menu_percentage_t0 = models.DecimalField(
+        null=True,
+        blank=True,
+        max_digits=5,
+        decimal_places=4,
+        validators=[MinValueValidator(0), MaxValueValidator(1)],
```

Comment:
This bug is so weird - when I made a test to see if I can give the API an empty string it didn't have the same behaviour as we saw in the browser...
When I removed the validators from the model and ran the migrations I still saw the error thrown from an empty string. So not sure why the valuesHt in diagnostic model don't have the same issue.
The error message is generated by IntegerField DRF serializer : venv/lib/python3.9/site-packages/rest_framework/fields.py (english message is 'A valid integer is required.')
So I went for a slightly hacky solution because I didn't want to spend yet more time trying to understand the issue. https://github.com/betagouv/ma-cantine/pull/1237/commits/20279dcd28f59636fe9c74ec7d709767decedc07